modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -378,6 +377,22 @@ void ONNXImporter::populateNet(Net dstNet)
     }
     dstNet.setInputsNames(netInputs);
 
+    // remove constant network inputs (initializers have already been added as constant nodes to graph)
+    int idx = 0;
+    while (graph_proto.input_size() != netInputs.size())
+    {
+        std::string name = graph_proto.input(idx).name();
+        if (std::find(netInputs.begin(), netInputs.end(), name) == netInputs.end())
+        {
+            graph_proto.mutable_input()->DeleteSubrange(idx, 1);
+        }
+        else
+        {
+            ++idx;
+        }
+    }
```

Comment:
Why not to remove only `Initializer` nodes at `addConstantNodesForInitializers`?

Reply:
`Initializers `overlap with inputs (not constant nodes). I think  removing `Initializers` is useless because we don't use initializers in `simplifySubgraphs`, only inputs and nodes.

Reply:
Because we have a procedure which replaces Initializers to Constants. That means other code which is related to this procedure should be also located there.